### PR TITLE
[B2BP-167] - Switch from dummy datas to CMS datas

### PR DESCRIPTION
**Pull request by Meolocious**
#### List of Changes
- Updated the "api.ts" file to align with the navigation plugin from Strapi.
- Replaced dummy data with real content from the CMS for the "getAllPages" constant used in the "generateStaticParams" function in the "page.tsx" file.

#### Motivation and Context
This change was necessary to integrate the project with real data from the CMS using the Strapi navigation plugin. Replacing dummy data with actual content improves the accuracy of the website's navigation and content display.

#### How Has This Been Tested?
The changes have been thoroughly tested in a development environment. We have verified that the "api.ts" file correctly fetches data from the CMS through the Strapi navigation plugin. Additionally, the "generateStaticParams" function in "page.tsx" functions as expected, using the real content provided by the updated "api.ts".

#### Types of changes
- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] Doesn't need any documentation updates

### CODE REVIEW

**Comment by Meolocious** [comment]
For the "api.ts" file instead, Simone helped me with matching the data from dummy to the navigation plugin data from CMS also to test navigation and error page if we go outside of the slugs generated.

For now the eslint is always in error due to some complication to fix the errors. This is a draft PR to check other things before any check on eslint part.

**Comment by Meolocious** [comment]
For running in local: rename ".env.local.example" in the root of "nextjs-website" in ".env.local"

File: apps/nextjs-website/src/lib/api.ts [hunk 1]
```diff
@@ -1,12 +1,124 @@
...
+    },
+  });
+
+  if (!response.ok) {
+    return {
+      error: 'Failed to fetch data from the API',
+    };
+  }
+
+  const navItems: ReadonlyArray<NavigationItem> = await response.json();
```
Comment (datalek) [inline apps/nextjs-website/src/lib/api.ts:119]: Add runtime type validation. E.g. using [io-ts](https://gcanti.github.io/io-ts/) or [zod](https://github.com/colinhacks/zod)

File: .infrastructure/03_network.tf
Comment (datalek) [inline .infrastructure/03_network.tf:1]: Revert this change

File: .infrastructure/06_alb.tf
Comment (datalek) [inline .infrastructure/06_alb.tf:1]: Revert this change

File: apps/nextjs-website/.env.example [hunk 2]
```diff
@@ -0,0 +1,2 @@
+NEXT_STRAPI_API_TOKEN=a7cc986b110df0aef1d499c2fbcdce5e6e5ca567127150a2dc00e2549c7b7960e73ede3425d676f04526e033a6f1227a9c1e0fa2269d4ff564a13ab52ce6ba8fce58d7d8b1298c42f1472ed7b83049b561c1c6e52b739a30dde44fd51bb3b8844b3e1ec35304e910a45dc2eaa0cd4d4a6486ebeb94a6078a78926c710866ed05
```
Comment (datalek) [inline apps/nextjs-website/.env.example:1]:
```suggestion
NEXT_STRAPI_API_TOKEN=aStrapiApiToken
```

File: apps/nextjs-website/src/lib/api.ts [hunk 3]
```diff
@@ -1,12 +1,124 @@
...
+}
+
+export interface Page {
+  readonly id: number;
+  readonly title: string;
+  readonly slug: ReadonlyArray<string>;
+  readonly visible: boolean;
+}
+
+const GeneratePagesFromNavItems = (
```
Comment (datalek) [inline apps/nextjs-website/src/lib/api.ts:35]:
```suggestion
const makePagesFromNavItems = (
```

File: apps/nextjs-website/tsconfig.json
Comment (datalek) [inline apps/nextjs-website/tsconfig.json:1]: Revert this chages

**Review by datalek — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/lib/api.ts [hunk 4]
```diff
@@ -1,12 +1,151 @@
...
-  // for now hardcoded, then retrieved from the cms
-  pages: [
-    { slug: ['a'] },
-    { slug: ['a', 'b'] },
-    { slug: ['a', 'b', 'c'] },
-    { slug: ['a', 'b', 'c', 'd'] },
-  ],
+import { z } from 'zod';
+
+// Zod schemas for validation
```
Comment (datalek) [inline apps/nextjs-website/src/lib/api.ts:3]: Do we need all these properties? If not can we just define the properties we need?

Comment (Meolocious) [reply apps/nextjs-website/src/lib/api.ts:3]: yes we need them all due to datas we'll get from the API calling. Without it, we cannot validate what we'll get and if there's some mismatch could be dangerous. Due to types needs to be declared we have only interface (or types) or zod method. In any of those cases, all those declarations are needed even if they seems to be a lot, 'cause the datas we're going to receive from the CMS are structured in this way and I don't suggest to avoid the data check.

**Comment by datalek** [comment]
Replaced by https://github.com/pagopa/b2b-portals/pull/61
